circuits/cpp/barretenberg/py/src/estimation/main.py

```diff
@@ -0,0 +1,71 @@
+from flavors import GoblinUltra
+from circuit import CircuitKZG
+from goblin import Goblin
+from protogalaxy import FoldingVerifier, Decider
+from utils import DataLog
+
+# App circuits an kernel circuits are in a "stack" and all
+# are assumed to have a fixed size
+LOG_N_IN_STACK = 13
+
+# A Goblin instance conists of three circuit instances, one for each proof output
+# by the client. A circuit instance contains the data of _estimates of_:
+#   - the size of the circuit 
+#   - the maximum amount of memory used during construction of the circuit 
+#   - the time to construct a recursive verifier circuit's MSMs using UltraCircuitConstructor
+#   - the time to construct a proof of a circuit of the given size
+#   - the size of the proof output by the circuit
+#
+# This data is logged using the logging class and then printed as a table
+
+
+class ClientStraightGoblin:
+    # create circuits to fold
+    def __init__(self, num_circuits):
+        self.num_circuits = num_circuits
+        circuits = [CircuitKZG(GoblinUltra(), log_n=LOG_N_IN_STACK, num_public_inputs=0)
+                    for _ in range(num_circuits)]
+        opqueue = []
+        for circuit in circuits[1:]:
+            opqueue += circuit.verifier_msms
+
+        self.goblin = Goblin(circuits[-1], opqueue)
+
+
+class ClientProtogalaxiedGoblin:
+    # create circuits to fold
+    def __init__(self, num_circuits):
+        self.num_circuits = num_circuits
+        circuits = [CircuitKZG(GoblinUltra(), log_n=LOG_N_IN_STACK, num_public_inputs=0)
+                    for _ in range(num_circuits)]
+        self.folding_verifier = FoldingVerifier(circuits)
+        self.decider = Decider(self.folding_verifier)
+
+        opqueue = []
+        opqueue += self.folding_verifier.msms
+        opqueue += self.folding_verifier.verifier_msms
+        self.goblin = Goblin(self.decider, opqueue)
+
+
+if __name__ == "__main__":
+    print("STRAIGHT GOBLIN")
+    data_log = DataLog()
+    for log_k in range(1, 11):
+        num_circuits = 1 << log_k
+        client_stack = ClientStraightGoblin(num_circuits)
+        data_log.add_entries(client_stack.goblin.final_circuit,
```

Comment:
do these data log entries have a "key"? i.e. a string that describes what is being logged. Could help a lot w.r.t. clarity and understanding the logs

Reply:
Yeah there is a struct that could be build, or we could use name initialization in the `add_entries` function for clarity.